plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php

```diff
@@ -401,54 +386,67 @@ private function register_post_type_for_order_placeholders(): void {
 	}
 
 	/**
-	 * Returns the HPOS setting for rendering in Features section of the settings page.
+	 * Add the definition for the HPOS feature.
 	 *
-	 * @param array  $feature_setting HPOS feature value as defined in the feature controller.
-	 * @param string $feature_id ID of the feature.
+	 * @param array $feature_definitions The array of feature definitions to add to.
 	 *
-	 * @return array Feature setting object.
+	 * @return array
 	 */
-	private function get_hpos_feature_setting( array $feature_setting, string $feature_id ) {
-		if ( ! in_array( $feature_id, array( self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION, DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION, 'custom_order_tables' ), true ) ) {
-			return $feature_setting;
-		}
-
-		if ( 'yes' === get_transient( 'wc_installing' ) ) {
-			return $feature_setting;
-		}
+	private function add_feature_definition( $feature_definitions ) {
+		$feature_definitions['custom_order_tables'] = array(
+			'name'                => __( 'High-Performance order storage', 'woocommerce' ),
+			'option_key'          => self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION,
+			'enabled_by_default'  => false,
+			'setting'             => $this->get_hpos_setting_for_feature(),
+			'additional_settings' => array(
+				$this->get_hpos_setting_for_sync(),
+			),
+		);
 
-		$sync_status = $this->data_synchronizer->get_sync_status();
-		switch ( $feature_id ) {
-			case self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION:
-				return $this->get_hpos_setting_for_feature( $sync_status );
-			case DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION:
-				return $this->get_hpos_setting_for_sync( $sync_status );
-			case 'custom_order_tables':
-				return array();
-		}
+		return $feature_definitions;
 	}
 
 	/**
 	 * Returns the HPOS setting for rendering HPOS vs Post setting block in Features section of the settings page.
 	 *
-	 * @param array $sync_status Details of sync status, includes pending count, and count when sync started.
-	 *
 	 * @return array Feature setting object.
 	 */
-	private function get_hpos_setting_for_feature( $sync_status ) {
-		$hpos_enabled            = $this->custom_orders_table_usage_is_enabled();
-		$plugin_info             = $this->features_controller->get_compatible_plugins_for_feature( 'custom_order_tables', true );
-		$plugin_incompat_warning = $this->plugin_util->generate_incompatible_plugin_feature_warning( 'custom_order_tables', $plugin_info );
-		$sync_complete           = 0 === $sync_status['current_pending_count'];
-		$disabled_option         = array();
-		// Changing something here? might also want to look at `enable|disable` functions in CLIRunner.
-		if ( count( array_merge( $plugin_info['uncertain'], $plugin_info['incompatible'] ) ) > 0 ) {
-			$disabled_option = array( 'yes' );
-		}
-		if ( ! $sync_complete ) {
-			$disabled_option = array( 'yes', 'no' );
+	private function get_hpos_setting_for_feature() {
+		if ( 'yes' === get_transient( 'wc_installing' ) ) {
```

Comment:
Why is this check necessary?

Reply:
I'm actually not sure. @vedanshujain this is from one of your commits, do you remember why this is here?